src/Jackett.Common/Indexers/InternetArchive.cs

```diff
@@ -203,5 +199,15 @@ private Uri GenerateMagnetLink(string btih, string title)
             _trackers.Set("dn", title);
             return new Uri("magnet:?xt=urn:btih:" + btih + "&" + _trackers.GetQueryString());
         }
+
+        private static string GetFieldAsString(JToken torrent, string field) =>
+            torrent[field] is JArray ?
+            (string)((JArray)torrent[field])[0] :
+            (string)torrent[field];
+
+        private static long GetFieldAsLong(JToken torrent, string field) =>
+            torrent[field] is JArray ?
+                (long)((JArray)torrent[field])[0] :
+                (long)torrent[field];
```

Comment:
Since these are essentially the same code, maybe they should be made generic? Also, ReSharper suggested pattern matching to avoid the cast on the other side of the `?` 

According to newtonsoft, this is how you would implement the Generic (naming could be better, but this is for illustration):

```C#
private static T GetFieldAs<T>(JToken torrent, string field) =>
            torrent[field] is JArray array
                ? array.First.ToObject<T>()
                : torrent.Value<T>(field);
```

Then use like `var title = GetFieldAs<string>(torrent, "title");` and `release.Size = GetFieldAs<long>(torrent, "item_size");` Keeps us from needing to add new functions for other types later.

Reply:
done